opensearch/src/main/java/org/opensearch/sql/opensearch/storage/system/OpenSearchSystemIndexEnumerator.java

```diff
@@ -25,10 +29,22 @@ public class OpenSearchSystemIndexEnumerator implements Enumerator<Object> {
 
   private ExprValue current;
 
-  public OpenSearchSystemIndexEnumerator(List<String> fields, OpenSearchSystemRequest request) {
+  /** Number of rows returned. */
+  private Integer queryCount;
+
+  /** ResourceMonitor. */
+  private final ResourceMonitor monitor;
+
+  public OpenSearchSystemIndexEnumerator(
+      List<String> fields, OpenSearchSystemRequest request, ResourceMonitor monitor) {
     this.fields = fields;
     this.request = request;
+    this.monitor = monitor;
+    this.queryCount = 0;
     this.current = null;
+    if (!this.monitor.isHealthy()) {
+      throw new NonFallbackCalciteException("insufficient resources to run the query, quit.");
+    }
```

Comment:
Healthy checking is missed in access OpenSearchSystemIndex.

Reply:
> I think we discussed this previously. Can we do this in non-intrusive way? Maybe AOP/wrap on operator level?

Yes, technically we could wrap it on operator level. But it will introduce
1. A new logical node `LogicalMonitor`
2. A `ConverterRule` to convert a logical node to physical node
3. A new physical node such as `EnumerableMonitor` similar to `ResourceMonitorPlan` in v2. 
4. Optimization #3853 has to find another way to workaround because the `root.rel` (physical plan) won't be a `Scannable` any more.

Not quite sure it's still worth to do the AOP way now.